### DISCUSSION TITLE
Why is navigation occasionally "stubborn"?

### DISCUSSION THREAD

**Discussion by livviest** [organicmaps/organicmaps#5847] [Q&A] [answered]
I live in a valley with two highways running along its length, one on each side. In OSM, one is marked as "primary" and the other as "secondary". When I plan a route from where I live to where I work the navigation will **greatly** favor the primary highway, which is on the other side of the valley. Even once I start down the secondary highway, it continues to reroute me to the primary highway via u-turn until much later when it finally "gives in" to my choice. This seems like undesirable behavior, especially since following these directions would add an additional 10min to my commute for seemingly no reason. Am I missing an option in the settings that can help me here?

**Comment by biodranik** [comment, marked as answer]
OM calculates the fastest route based on the data from osm.org (ignoring traffic lights and turns penalties at the moment). The main question is why the route is not the fastest. Is it properly mapped at osm.org ? Is it because turns and traffic lights are ignored now? Is it because of the traffic information (that is absent in OM)?

**Reply by livviest** [reply]
Thanks for the swift response! What fields from OSM are used to calculate the fastest route in OM? Maybe if I know this I can find out if there's something wonky about the OSM data for the highways in my area (and fix it).

**Reply by biodranik** [reply]
road type, surface, speed limit
